retry.go

```diff
@@ -75,7 +75,9 @@ func Retry[T any](ctx context.Context, operation Operation[T], opts ...RetryOpti
 
 	// Apply user-provided options to the default settings.
 	for _, opt := range opts {
-		opt(args)
+		if opt != nil {
```

Comment:
I don't think that we should tolerate a nil value here. It's atypical in functional option-consuming functions or methods to do so. Since the option type is exported, callers can build up a slice of such options incrementally and conditionally before calling on this function.

Reply:
You are right